sway-core/src/semantic_analysis/namespace/module.rs

```diff
@@ -52,110 +51,99 @@ pub struct Module {
+                assert!(warnings.is_empty());
 
-            // Invariant: `.value == None` => `!errors.is_empty()`.
-            vec1::Vec1::try_from_vec(errors).unwrap()
-        })
+                // Invariant: `.value == None` => `!errors.is_empty()`.
+                vec1::Vec1::try_from_vec(errors).unwrap()
+            },
+        )
     }
 
-    fn default_with_constants_inner(
+    fn default_with_contract_id_inner(
         handler: &Handler,
         engines: Engines<'_>,
-        constants: BTreeMap<String, ConfigTimeConstant>,
         ns_name: Option<Ident>,
+        contract_id_value: String,
     ) -> Result<Self, ErrorEmitted> {
         // it would be nice to one day maintain a span from the manifest file, but
         // we don't keep that around so we just use the span from the generated const decl instead.
         let mut compiled_constants: SymbolMap = Default::default();
         // this for loop performs a miniature compilation of each const item in the config
-        for (
-            name,
-            ConfigTimeConstant {
-                r#type,
-                value,
-                public,
-            },
-        ) in constants.into_iter()
-        {
-            // FIXME(Centril): Stop parsing. Construct AST directly instead!
-            // parser config
-            let const_item = match public {
-                true => format!("pub const {name}: {type} = {value};"),
-                false => format!("const {name}: {type} = {value};"),
-            };
-            let const_item_len = const_item.len();
-            let input_arc = std::sync::Arc::from(const_item);
-            let token_stream = lex(handler, &input_arc, 0, const_item_len, None).unwrap();
-            let mut parser = Parser::new(handler, &token_stream);
-            // perform the parse
-            let const_item: ItemConst = parser.parse()?;
-            let const_item_span = const_item.span().clone();
-
-            // perform the conversions from parser code to parse tree types
-            let name = const_item.name.clone();
-            let attributes = Default::default();
-            // convert to const decl
-            let const_decl = to_parsed_lang::item_const_to_constant_declaration(
-                &mut to_parsed_lang::Context::default(),
-                handler,
-                engines,
-                const_item,
-                attributes,
-                true,
-            )?;
-
-            // Temporarily disallow non-literals. See https://github.com/FuelLabs/sway/issues/2647.
-            let has_literal = match &const_decl.value {
-                Some(value) => {
-                    matches!(value.kind, ExpressionKind::Literal(_))
-                }
-                None => false,
-            };
+        // FIXME(Centril): Stop parsing. Construct AST directly instead!
+        // parser config
+        let const_item = format!("pub const CONTRACT_ID: b256 = {contract_id_value};");
```

Comment:
This line is probably the only notable change in this entire block, but the unindentation (since we're no longer looping through a bunch of constants) is displaying this entire block as diff'd